src/app/tweet/tweet.component.spec.ts

```diff
@@ -1,19 +1,20 @@
-import { Observable, of } from 'rxjs';
+import { NEVER, Observable, of } from 'rxjs';
 import { TestScheduler } from 'rxjs/testing';
+
 import { TweetComponent } from './tweet.component';
 import { TweetService } from './tweet.service';
 
+jest.mock('./tweet.service');
+
 describe('TweetComponent', () => {
   let testScheduler: TestScheduler;
 
-  let tweetService: jasmine.SpyObj<TweetService>;
+  let tweetService: jest.Mocked<TweetService>;
 
   beforeEach(() => {
     testScheduler = new TestScheduler((actual, expected) => expect(actual).toEqual(expected));
 
-    tweetService = jasmine.createSpyObj('TweetService', [
-      'getLikeCount', 'getRetweetCount', 'getReplyCount',
-    ]);
+    tweetService = new TweetService() as jest.Mocked<TweetService>;
```

Comment:
This example is a bit confusing even to me still because it is sorta misleading to the developer. The line `jest.mock('./tweet.service')` is what takes `TweetService` and creates a new class with a zero-arg constructor. This is convenient because it allows you to create an instance without needing to provide the constructor args, but this is not obvious because the TS complier still sees this as the original class. Looking into ways to do this in a more type safe way.